recipes/cpython/all/conanfile.py

```diff
@@ -0,0 +1,720 @@
+            if self.settings.os != "Windows":
+                if not tools.is_apple_os(self.settings.os):
+                    self.cpp_info.components["_hidden"].requires.append("libuuid::libuuid")
+                self.cpp_info.components["_hidden"].requires.append("libxcrypt::libxcrypt")
+            if self.options.with_bz2:
+                self.cpp_info.components["_hidden"].requires.append("bzip2::bzip2")
+            if self.options.get_safe("with_gdbm", False):
+                self.cpp_info.components["_hidden"].requires.append("gdbm::gdbm")
+            if self.options.with_sqlite3:
+                self.cpp_info.components["_hidden"].requires.append("sqlite3::sqlite3")
+            if self.options.get_safe("with_curses", False):
+                self.cpp_info.components["_hidden"].requires.append("ncurses::ncurses")
+            if self.options.get_safe("with_bsddb"):
+                self.cpp_info.components["_hidden"].requires.append("libdb::libdb")
+            if self.options.get_safe("with_lzma"):
+                self.cpp_info.components["_hidden"].requires.append("xz_utils::xz_utils")
+            if self.options.get_safe("with_tkinter"):
+                self.cpp_info.components["_hidden"].requires.append("tk::tk")
+            self.cpp_info.components["_hidden"].libdirs = []
+
+        bindir = os.path.join(self.package_folder, "bin")
+        self.output.info("Appending PATH environment variable: {}".format(bindir))
+        self.env_info.PATH.append(bindir)
+
+        python = self._cpython_interpreter_path
+        self.user_info.python = python
+        if self.options.env_vars:
+            self.output.info("Setting PYTHON environment variable: {}".format(python))
+            self.env_info.PYTHON = python
+
+        if self.settings.compiler == "Visual Studio":
+            pythonhome = os.path.join(self.package_folder, "bin")
+        else:
+            version = tools.Version(self._version_number_only)
+            pythonhome = os.path.join(self.package_folder, "lib", "python{}.{}".format(version.major, version.minor))
+        self.user_info.pythonhome = pythonhome
+
+        pythonhome_required = self.settings.compiler == "Visual Studio" or tools.is_apple_os(self.settings.os)
+        self.user_info.module_requires_pythonhome = pythonhome_required
+
+        if self.settings.compiler == "Visual Studio":
+            if self.options.env_vars:
+                self.output.info("Setting PYTHON environment variable: {}".format(pythonhome))
+                self.env_info.PYTHONHOME = pythonhome
+
+        if self._is_py2:
+            python_root = ""
+        else:
+            python_root = self.package_folder
+            if self.options.env_vars:
+                self.output.info("Setting PYTHON_ROOT environment variable: {}".format(python_root))
+                self.env_info.PYTHON_ROOT = python_root
+        self.user_info.python_root = python_root
```

Comment:
@SSE4 
The current fails are due to wrong environment variables. I haven't found the right combination yet.
It's something to do with `PYTHONHOME` and/or `PYTHON_ROOT`
In my latest commits I was messing a bit with them. Since then, I took a break from this recipe.

2.7 has a different behavior as 3.X
Windows has a different behavior as Linux
And Macos is also causing problems

I haven't found the golden combination yet, as I don't own a Mac.
It would be nice if you could pinpoint the combination, so we can fix it (and document it).

Also, 3.10 will probably be released this month.

Reply:
okay, I'll try to continue from where you did a break, I hope you don't mind :)

Reply:
Please do!!!!
I know all combinations build.
It's only the tests that fail.
Because the time between git push and c3i results is so long, I postponed it.